pointgrey_camera_driver/src/custom_stereo.cpp

```diff
@@ -0,0 +1,625 @@
+            left_cam_.disconnect();
+            right_cam_.disconnect();
+            NODELET_INFO("Disconnected from camera.");
+
+            state = DISCONNECTED;
+          }
+          catch(std::runtime_error& e)
+          {
+            NODELET_ERROR_COND(state_changed,
+                "Failed to disconnect with error: %s", e.what());
+            ros::Duration(1.0).sleep(); // sleep for one second each time
+          }
+
+          break;
+        case DISCONNECTED:
+          // Try connecting to the camera
+          try
+          {
+            NODELET_DEBUG("Connecting to camera.");
+            left_cam_.connect();
+            right_cam_.connect();
+            NODELET_INFO("Connected to camera.");
+
+            // Set last configuration, forcing the reconfigure level to stop
+            left_cam_.setNewConfiguration(config_, PointGreyCamera::LEVEL_RECONFIGURE_STOP);
+            right_cam_.setNewConfiguration(config_, PointGreyCamera::LEVEL_RECONFIGURE_STOP);
+
+            // Set the timeout for grabbing images.
+            try
+            {
+              double timeout;
+              getMTPrivateNodeHandle().param("timeout", timeout, 1.0);
+
+              NODELET_DEBUG("Setting timeout to: %f.", timeout);
+              left_cam_.setTimeout(timeout);
+              right_cam_.setTimeout(timeout);
+            }
+            catch(std::runtime_error& e)
+            {
+              NODELET_ERROR("%s", e.what());
+            }
+
+            // Subscribe to gain and white balance changes
+            {
+              boost::mutex::scoped_lock scopedLock(connect_mutex_);
+              sub_ = getMTNodeHandle().subscribe("image_exposure_sequence", 10, &pointgrey_camera_driver::PointGreyCustomStereoNodelet::gainWBCallback, this);
+            }
+
+            state = CONNECTED;
+          }
+          catch(std::runtime_error& e)
+          {
+            NODELET_ERROR_COND(state_changed,
+                "Failed to connect with error: %s", e.what());
+            ros::Duration(1.0).sleep(); // sleep for one second each time
+          }
+
+          break;
+        case CONNECTED:
+          // Try starting the camera
+          try
+          {
+            NODELET_DEBUG("Starting camera.");
+            left_cam_.start();
+            right_cam_.start();
+            NODELET_INFO("Started camera.");
+
+            state = STARTED;
+          }
+          catch(std::runtime_error& e)
+          {
+            NODELET_ERROR_COND(state_changed,
+                "Failed to start with error: %s", e.what());
+            ros::Duration(1.0).sleep(); // sleep for one second each time
+          }
+
+          break;
+        case STARTED:
+          {
+            // Trigger cameras if using software triggering
+            if (config_.enable_trigger == true && config_.trigger_source == "software")
+            {
+              if (!left_cam_.trigger() || !right_cam_.trigger())
```

Comment:
optional: Maybe split in two, so you can mentioned in the error message which camera (left or right) couldn't be triggered.